controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -108,14 +109,17 @@ else if (report.jobType().isProduction() && deploymentComplete(application)) {
             // TODO jvenstad: Don't trigger.
             // Trigger next
             if (report.success()) {
-                List<JobType> jobs = order.nextAfter(report.jobType(), application);
-                for (JobType job : jobs)
-                     application = trigger(new Triggering(application, job, false, report.jobType().jobName() + " completed"), jobs, false);
+                triggerReadyJobs(application);
+                return; // Don't overwrite below.
+            }
+            else if (retryBecauseOutOfCapacity(application, report.jobType())) {
+                triggerReadyJobs(application);
+                return; // Don't overwrite below.
+            }
+            else if (retryBecauseNewFailure(application, report.jobType())) {
+                triggerReadyJobs(application);
+                return; // Don't overwrite below.
```

Comment:
It would be great if we could store in only one place...

Reply:
Yes! I'll work towards that.